components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/service/impl/HTTPEventPublisherImpl.java

```diff
@@ -0,0 +1,146 @@
+            throw new EventPublisherServerException(ERROR_ACTIVE_WEBHOOKS_NOT_FOUND.getMessage(),
+                    ERROR_ACTIVE_WEBHOOKS_NOT_FOUND.getDescription(), ERROR_ACTIVE_WEBHOOKS_NOT_FOUND.getCode(), e);
+        }
+    }
+
+    private void makeAsyncAPICall(SecurityEventTokenPayload eventPayload, EventContext eventContext)
+            throws HTTPAdapterException {
+
+        ClientManager clientManager = HTTPAdapterDataHolder.getInstance().getClientManager();
+
+        for (Webhook webhook : activeWebhooks) {
+            String url = webhook.getEndpoint();
+            String secret = webhook.getSecret();
+
+            HttpPost request = clientManager.createHttpPost(url, eventPayload, secret);
+
+            logPublishingEvent(url, eventContext, eventContext.getEventUri());
+
+            final long requestStartTime = System.currentTimeMillis();
+
+            CompletableFuture<HttpResponse> future = clientManager.executeAsync(request);
+
+            future.thenAccept(response -> handleAsyncResponse(
+                            response, request, requestStartTime, eventContext, url, eventContext.getEventUri()))
+                    .exceptionally(ex -> {
+                        handleResponseCorrelationLog(request, requestStartTime,
+                                HTTPCorrelationLogUtils.RequestStatus.FAILED.getStatus(),
+                                ex.getMessage());
+                        log.error("Publishing event data to HTTP failed. ", ex);
+                        throw new IdentityRuntimeException("Error occurred while publishing event data to HTTP. ", ex);
+                    });
+        }
+    }
+
+    private static void handleAsyncResponse(HttpResponse response, HttpPost request, long requestStartTime,
+                                            EventContext eventContext, String url, String eventUri) {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(eventContext.getTenantDomain());
+
+        try {
+            int responseCode = response.getStatusLine().getStatusCode();
+            String responsePhrase = response.getStatusLine().getReasonPhrase();
+            log.debug("HTTP request completed. Response code: " + responseCode);
+
+            handleResponseCorrelationLog(request, requestStartTime,
+                    HTTPCorrelationLogUtils.RequestStatus.COMPLETED.getStatus(),
+                    String.valueOf(responseCode), responsePhrase);
+            if (responseCode >= 200 && responseCode < 300) {
+                logDiagnosticSuccess(eventContext, url, eventUri);
+                log.debug("HTTP request completed. Response code: " + responseCode);
+            } else {
+                logDiagnosticFailure(eventContext, url, eventUri);
+                log.debug("HTTP request failed with response code: " + responseCode +
+                        " and reason phrase: " + responsePhrase);
```

Comment:
Can we have a better term than reason phrase ? May be you can say 'due to'

Reply:
fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/37/commits/15538dc2b9ce588c47e8db9c5db39156de457b3c